src/traces/mesh3d/defaults.js

```diff
@@ -34,20 +34,13 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     }
 
     var coords = readComponents(['x', 'y', 'z']);
-    var indices = readComponents(['i', 'j', 'k']);
+    readComponents(['i', 'j', 'k']);
 
     if(!coords) {
         traceOut.visible = false;
         return;
     }
 
-    if(indices) {
-        // otherwise, convert all face indices to ints
-        indices.forEach(function(index) {
-            for(var i = 0; i < index.length; ++i) index[i] |= 0;
```

Comment:
Hmm. So you didn't move this to `mesh3d/convert.js`? Just :hocho: ing it didn't break any tests?

Can you try plotting a mesh3d trace with non-integer indices and see what happens?

Reply:
@etpinard Thanks for the review. 
That's true! One could get bad errors in case of having non-integer ids. 
So, in the revised code I added map functions to round the float values to integers. 
However; there is now extra logic to check and handle for such cases. A series of jasmine tests are also added to define the expected behaviour in various scenarios.